### ISSUE TITLE
acme.sh badCSR SAN Invalid Error

### ISSUE THREAD

**Issue by tjmullicani** [glatzert/ACME-Server-ADCS#14]
I was testing with acme.sh and noticed there is an error caused by https://github.com/acmesh-official/acme.sh/issues/1335, https://github.com/acmesh-official/acme.sh/commit/f8ca6d98337a2eb1dbfcde8f730241eb46884a94.

Due to acme.sh adding the `extendedKeyUsage` extension to the automatically generated CSR, the `CsrValidator.cs SubjectAlternateNamesAreValid` function returns false since `x509Ext.ObjectId.Name` is now equal to `XCN_OID_ENHANCED_KEY_USAGE` first. 
https://github.com/glatzert/ACME-Server-ADCS/blob/3b45df312c51050d7366c58a663b2cef9c2823d4/src/ACME.CertProvider.ADCS/CsrValidator.cs#L88

Removing `extendedKeyUsage=serverAuth,clientAuth` from the acme.sh script (part that generates the CSR file) fixes the issue. Not sure if this needs to be addressed here or with acme.sh.

```console
[Thu 06 Apr 2023 04:50:20 AM UTC] Sign error, wrong status
[Thu 06 Apr 2023 04:50:20 AM UTC] {"status":"invalid","identifiers":[{"type":"dns","value":"test.example.local"}],"error":{"type":"urn:ietf:params:acme:error:badCSR","detail":"SAN Invalid."},"authorizations":["https://caissueadcs.example.local/order/RqaBOEM7pUG0wnxYNc_INg/auth/Cpi4l4UtRkCiiJaoTd5nFA"],"finalize":"https://acme.example.local/order/RqaBOEM7pUG0wnxYNc_INg/finalize"}
```

**Comment by glatzert** [comment]
I'll fix it in the server.
It's a oversight of me to not allow other extensions besides SAN.

**Comment by glatzert** [comment]
Most likely fixed in V1.0.3 (I did not run tests yet)

**Comment by tjmullicani** [comment]
@glatzert V1.0.3 fixed the issue
